include/seqan3/io/alignment_file/output.hpp

```diff
@@ -0,0 +1,706 @@
+ *                              fields IDs; only relevant if these can't be deduced.
+ * \tparam valid_formats        A seqan3::type_list of the selectable formats (each
+ *                              must model seqan3::alignment_file_output_format_concept).
+ * \tparam stream_type          The type of the stream, must model seqan3::ostream_concept.
+ *
+ * \details
+ *
+ * ### Introduction
+ *
+ * Alignment files provide a way to store pairwise alignment information.
+ * Well-known formats include SAM and BAM or BLAST.
+ *
+ * The Alignment file abstraction supports writing following fields:
+ *
+ *   1. field::SEQ
+ *   2. field::ID
+ *   3. field::OFFSET
+ *   4. field::REF_SEQ
+ *   5. field::REF_ID
+ *   6. field::REF_OFFSET
+ *   7. field::ALIGNMENT
+ *   8. field::MAPQ
+ *   9. field::FLAG
+ *   10. field::QUAL
+ *   11. field::MATE
+ *   12. field::TAGS
+ *   13. field::EVALUE
+ *   14. field::BIT_SCORE
+ *
+ * There is an additional field called `field::HEADER_PTR`, it used to transfer
+ * header information from `alignment_file_input` to `alignment_file_output`,
+ * but you needn't deal with this field manually.
+ *
+ * The member functions take any and either of these fields.
+ *
+ * ### Construction and specialisation
+ *
+ * This class comes with two constructors, one for construction from a file name
+ * and one for construction from an existing stream and a known format. The first
+ * one automatically picks the format based on the extension of the file name.
+ * The second can be used if you have a non-file stream, like std::cout or
+ * std::ostringstream, that you want to read from and/or if you cannot use
+ * file-extension based detection, but know that your output file has a certain
+ * format.
+ *
+ * In most cases the template parameters are deduced completely automatically:
+ *
+ * \snippet test/snippet/io/alignment_file/output.cpp filename_construction
+ *
+ * Writing to std::cout:
+ *
+ * \snippet test/snippet/io/alignment_file/output.cpp format_construction
+ *
+ * Note that this is not the same as writing `alignment_file_output<>`
+ * (with angle brackets). In the latter case they are explicitly set to their
+ * default values, in the former case
+ * [automatic deduction](http://en.cppreference.com/w/cpp/language/class_template_argument_deduction)
+ * happens which chooses different parameters depending on the constructor arguments.
+ * For opening from file, `alignment_file_output<>` would have also worked, but for
+ * opening from stream it would not have.
+ *
+ * ### Writing record-wise
+ *
+ * \snippet test/snippet/io/alignment_file/record_based_writing.cpp all
+ *
+ * The easiest way to write to an alignment file is to use the push_back() member functions. These
+ * work similarly to how they work on an std::vector. If you pass a tuple to push_back() the seqan3::field ID
+ * of the i-th tuple-element/argument is assumed to be the i-th value of selected_field_ids.
```

Comment:
like I said before, please point out that push_back(record) should be used and not the tuple interface.